### ISSUE TITLE
Discrepancy between pol and Sz

### ISSUE THREAD

**Issue by DillonBus** [wschreyer/PENTrack#58]
I noticed that in a uniform z-oriented field where the material containing the UCN has a 100% spin flip probability, that the result of Sz is not changing sign when the direction of polarization changes from pol1=1 to pol2=-1 during a collision. 

The following link is a figure of this behavior demonstrating this behavior.
![PolSzDisagree](https://github.com/wschreyer/PENTrack/assets/164058594/cdf410f7-6703-48e9-84b1-069398d44c70) 

I am unsure if this is a bug or if the actual spin should be interpreted as the product of the spin and polarization (polend*Szend). Thank you for looking into this and I apologize if it is just a misunderstanding on my part!

**Comment by wschreyer** [comment]
Hi. That's a very peculiar situation (the UCN hits a 100% spin-flipping surface 28 times within a second?).
It seems the Sz sample points coincide with the surface hits, where the polarization is ambiguous (+1 and -1 at the same time). The few sample points that don't coincide with surface hits happen to fall into periods where the polarization is +1. So it's not entirely clear if there actually is an issue.
You could try reducing the time between Sz samples to get more samples in between surface hits. There currently is no way of making the sampling interval smaller than the interval chosen by the motion integrator, but you could increase the magnetic field, which should naturally shorten the integration intervals.

**Comment by DillonBus** [comment]
You are right, the simulation is "UCN hits a 100% spin-flipping surface 28 times within a second." 

As you suggested, I increased the field to reduce the interval of Spinlog outputs and am observing the same behavior (shown below).

![PolSzDisagreeRamsey2](https://github.com/wschreyer/PENTrack/assets/164058594/ecf5bb00-0c38-49f2-8c49-31fb3c7e4b24)

The reason I first noticed this discrepancy was in a simulation where I let UCNs in a uniform field bounce around a cell with dPS coated walls and NiP coated electrodes. I was expecting the material properties of the cells to cause spin flips. In the EndLog of this simulation, I saw the following. 

<img width="322" alt="Screenshot 2024-05-20 at 8 27 57 AM" src="https://github.com/wschreyer/PENTrack/assets/164058594/dc242193-cf03-4657-b499-4ccbc6f4ab40">


Even though some of the particles have a spin flip (Nspinflip=1 and polend=1), the spin never changes (Szend=-1). It also says that the spinflipprob=0, but I assume this is the probability is zero since the field is uniform and this value comes from the integration of the BMT equation.

**Comment by wschreyer** [comment]
You were right, the spin flips during wall hits were not applied to the spin tracking. I committed some changes that during my first tests seem to result in the right behavior. It would be great if you could give it a try and report if you see the behavior you expect. Thank you!

**Comment by DillonBus** [comment]
It appears to be working correctly now!
![PolSzAgree](https://github.com/wschreyer/PENTrack/assets/164058594/7eeeba4a-9a9e-4b88-99fd-13ec0bd59d02)
